flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableSink.java

```diff
@@ -161,4 +164,37 @@ private String toStagingDir(String finalDir, Configuration conf) throws IOExcept
 		fs.deleteOnExit(path);
 		return res;
 	}
+
+	@Override
+	public List<String> getPartitionFieldNames() {
+		return catalogTable.getPartitionKeys();
+	}
+
+	@Override
+	public void setStaticPartition(Map<String, String> partitions) {
+		// make it a LinkedHashMap to maintain partition column order
+		staticPartitionSpec = new LinkedHashMap<>();
+		for (String partitionCol : getPartitionFieldNames()) {
+			if (partitions.containsKey(partitionCol)) {
+				staticPartitionSpec.put(partitionCol, partitions.get(partitionCol));
+			}
+		}
+	}
+
+	private void validatePartitionSpec() {
+		List<String> partitionCols = getPartitionFieldNames();
+		Preconditions.checkArgument(new HashSet<>(partitionCols).containsAll(
```

Comment:
A few comments here:

- can we output which column is unknown to the error message?
- the check of `new HashSet<>(partitionCols).containsAll(staticPartitionSpec.keySet())` will lose order of the partitions, which seems to be not enough for a legit validation. Shall it be `partitionCols.equals(new ArrayList(staticPartitionSpec.keySet()))`? since staticPartitionSpec is a linked hashmap, keys from its `keySet()` should be ordered

- nit: can we reformat it to be more readable?

```
Preconditions.checkArgument(
    ...
    "...";
```

Reply:
Yeah I'll print the specific columns and reformat the code.
We don't need the check the order here. Partition columns order is defined by `getPartitionFieldNames()`. We can always reorder a partition spec (which is a map) as long as it only contains valid partition columns.